src/IO/mesh/fortran/read_interfaces.cpp

```diff
@@ -0,0 +1,86 @@
+#include "IO/mesh/fortran/read_interfaces.hpp"
+#include "IO/fortranio/interface.hpp"
+#include "mesh/coupled_interfaces/coupled_interfaces.hpp"
+#include "mesh/coupled_interfaces/interface_container.hpp"
+#include "specfem_mpi/interface.hpp"
+
+template <specfem::element::medium_tag medium1,
+          specfem::element::medium_tag medium2>
+specfem::mesh::interface_container<medium1, medium2>
+specfem::IO::mesh::fortran::read_interfaces(const int num_interfaces,
+                                            std::ifstream &stream,
+                                            const specfem::MPI::MPI *mpi) {
+
+  specfem::mesh::interface_container<medium1, medium2> interface(
+      num_interfaces);
+
+  if (!num_interfaces)
+    return interface;
+
+  int medium1_ispec_l, medium2_ispec_l;
+
+  for (int i = 0; i < num_interfaces; i++) {
+    specfem::IO::fortran_read_line(stream, &medium2_ispec_l, &medium1_ispec_l);
+    interface.medium1_index_mapping(i) = medium1_ispec_l - 1;
+    interface.medium2_index_mapping(i) = medium2_ispec_l - 1;
+  }
+
+  return interface;
+}
+
+// Explicit instantiation of the template function for the different medium
+// interfaces elastic/acoustic
+template specfem::mesh::interface_container<
+    specfem::element::medium_tag::elastic,
+    specfem::element::medium_tag::acoustic>
+specfem::IO::mesh::fortran::read_interfaces<
```

Comment:
You dont nacessarily need this explicit instantiation here, since, this function is only ever used by `read_coupled_interfaces`. `read_coupled_interfaces` knows the definition from above function and wouldn't use the explicit instantiation I believe.